### DISCUSSION TITLE
视频流返回前端无法播放

### DISCUSSION THREAD

**Discussion by liyang9331** [eggjs/egg#4767] [General]
ctx.set('content-type', 'video/mp4');//设置返回类型
    let stream = fs.createReadStream('./assets/test.mp4');
    if (stream) {//判断状态
      stream.on('open', function () {
        console.log('文件打开')
      });
      stream.on('end', function () {
        console.log("读取完毕")
      })
      stream.on('data', function (data) {
        // console.log("读取文件流")
        ctx.body = data
      })
      stream.on('err', function () {
        console.log("发生错误")
      })
      stream.on('close', function () {
        console.log('关闭')
        // ctx.end
      });
    }

**Comment by whatwewant** [comment]
only => `ctx.body = stream`
